src/app/(auth)/login/page.tsx

```diff
@@ -7,36 +7,60 @@ import TextInput from '@/components/TextInput/index';
 import { H1, P, LinkColored } from '@/styles/text';
 import supabase from '@/api/supabase/createClient';
 import COLORS from '@/styles/colors';
-import { H4Centered, QuestionsDiv, SpacerDiv } from '@/app/(auth)/styles';
+import { H4Centered, SpacerDiv } from '@/app/(auth)/styles';
 import BigButton from '@/components/BigButton';
 
 export default function Login() {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  // const [errorMessage, setErrorMessage] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
   const { push } = useRouter();
 
   const handleSignIn = async () => {
+    if (email === '' || password === '') {
+      if (email === '') {
+        setEmailError('Invalid Email');
+      } else {
+        setEmailError('');
+      }
+      if (password === '') {
+        setPasswordError('Invalid Password');
+      } else {
+        setPasswordError('');
+      }
+      setErrorMessage('');
+      return;
+    }
+    setEmailError('');
+    setPasswordError('');
+
     const { error } = await supabase.auth.signInWithPassword({
       email,
       password,
     });
 
     if (error) {
-      throw new Error(`An error occurred trying to sign in: ${error.message}`);
+      setErrorMessage(error.message);
+    } else {
+      setErrorMessage('');
+      push('/');
     }
-    push('/');
   };
 
   return (
     <>
-      <H1>Log In</H1>
-      <SpacerDiv>
-        <QuestionsDiv>
+      <SpacerDiv $gap={0.625}>
+        <H1>Log In</H1>
+        {errorMessage !== '' && <P $color={COLORS.redMid}>{errorMessage}</P>}
+      </SpacerDiv>
+      <SpacerDiv $gap={13 / 16}>
```

Comment:
nit: can we turn this into a single number?

Reply:
done!